File: Programmers/Level1/약수의개수와덧셈/약수의개수와덧셈.js
function getDivisorCnt(num) {
  const ret = [];
  for (let i = 1; i < Math.sqrt(num); i++) {
    if (!(num % i)) {
      ret.push(i);
      ret.push(num / i);
    }
  }
  if (!(num % Math.sqrt(num))) {
    ret.push(Math.sqrt(num));
  }

  return ret.length;
}

function solution(left, right) {
  let total = 0;
  for (let i = left; i <= right; i++) {
    const cnt = getDivisorCnt(i);
    cnt % 2 ? (total -= i) : (total += i);
  }
  return total;
}

const left = 13;
const right = 17;

console.log(solution(left, right));

File: Programmers/Level1/약수의개수와덧셈/약수의개수와덧셈.py
def get_divisor_cnt(num):
    ret = []
    for i in range(1, int(num ** 0.5) + 1):
        if not num % i:
            ret.append(i)
            ret.append(num // i)

    return len(ret) if num % num**0.5 else len(ret) + 1


def solution(left, right):
    answer = 0
    for i in range(left, right+1):
        cnt = get_divisor_cnt(i)
        answer = answer - i if cnt % 2 else answer + i
    return answer


left = 13
right = 17
print(solution(left, right))
